bundles/tools.vitruv.domains.java.ui/src/tools/vitruv/domains/java/ui/monitorededitor/JavaMonitoredEditor.xtend

```diff
@@ -126,14 +134,40 @@ class JavaMonitoredEditor extends AbstractMonitoredEditor implements ChangeOpera
 							" was not found although it should exist")
 					}
 				}
-			try {
-				this.virtualModel.propagateChangedState(changedResource, change.oldResourceURI)
+			//TODO This is just a workaround to access the previous resource state.
+			// The correct behavior would be to attach the Java monitor to a state-deriving view and reuse its change deriving functionality.
+			try (val view = getViewForURI(change.oldResourceURI ?: change.newResourceURI)) {
+			    val oldResource = view.rootObjects.head?.eResource
+			    val changeSequence = generateChange(changedResource, oldResource)
+			    virtualModel.propagateChange(changeSequence)
 			} catch (Exception e) {
 				log.error('''Some error occurred during propagating changes in projects «monitoredProjectNames»''', e)
 				throw new IllegalStateException(e)
 			}
 		}
 	}
+	
+	private def View getViewForURI(URI resourceURI) {
+	    val selector = virtualModel.createSelector(ViewTypeFactory.createIdentityMappingViewType("Java"))
+
+        for (rootElement : selector.selectableElements.filter[it.eResource?.URI == resourceURI]) {
+            selector.setSelected(rootElement, true)
+        }
+        return selector.createView()
```

Comment:
The formatting does not seem to be proper.

Reply:
normalized in [7492187](https://github.com/vitruv-tools/Vitruv-Domains-ComponentBasedSystems/pull/113/commits/74921874311246101c2d2de540eacc135a6a44bb)